sdk/core/core-http/src/webResource.ts

```diff
@@ -37,6 +37,49 @@ export type TransferProgressEvent = {
   loadedBytes: number;
 };
 
+export interface WebResourceLike {
```

Comment:
Any way you could use some nifty typing tricks to inherit and manipulate the properties from WebResource (and same for HttpHeaders)?  That would relieve us from keeping them in sync if we ever change them again.  Probably not possible but I figured it wouldn't hurt to ask!

Reply:
I don't think we'd really want to since it's intentional that the concrete types could have extra things (like `TestSpan` implementing `Span` for tracing.)

In theory it shouldn't bite us because if you need to access a new prop (say in a policy) you'd have to add it to the interface first anyway.